core/src/main/java/feign/Response.java

```diff
@@ -121,12 +122,13 @@ public Builder body(String text, Charset charset) {
 
     /**
      * @see Response#request
-     *
-     *      NOTE: will add null check in version 10 which may require changes to custom feign.Client
-     *      or loggers
      */
     public Builder request(Request request) {
-      this.request = request;
+      checkNotNull(request, "the original request is required on all responses");
+
+      /* don't keep the body, we don't want to tie up memory on large requests */
+      this.request = Request.create(
```

Comment:
My concern is with larger requests that deal with multi-part uploads and pure binary requests.  Keeping the body in memory after the request feels like it may lead to unexpected memory usages and leaks.  I had not considered the GC impact.

Reply:
It feels like an early optimization for me.

What if we include the original `Request` for now, and try to fix the problem when we spot real use cases?

Reply:
Understood.  I'll change to to reference the original and not create a new request.